### PR TITLE
Remove results buffering and drop BufferredStatement.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
Not sure why `rowCount()` fails after an INSERT query for SqlServer. I tried adding scrollable cursor option for the statement just in case but doesn't help.

```
1) Cake\Test\TestCase\ORM\QueryTest::testInsert
Failed asserting that -1 matches expected 2.

D:\a\cakephp\cakephp\tests\TestCase\ORM\QueryTest.php:1748
```

File: src/Database/Driver/Sqlserver.php [hunk 1]
```diff
@@ -191,10 +191,6 @@ public function prepare(Query|string $query): StatementInterface
         $this->connect();
 
         $sql = $query;
-        $options = [
-            PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL,
-            PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE => PDO::SQLSRV_CURSOR_BUFFERED,
```
Comment (othercorey) [inline src/Database/Driver/Sqlserver.php:196]: We still want this. This makes sqlsrv driver fetch the results instead of using a server-side buffer.

Comment (ADmad) [reply src/Database/Driver/Sqlserver.php:196]: Restored below, all green now.

File: src/Database/Query.php [hunk 2]
```diff
@@ -2168,60 +2160,6 @@ public function setValueBinder(?ValueBinder $binder)
...
-     * buffered. This enables you to iterate a result set multiple times, or
-     * both cache and iterate it.
-     *
-     * When disabled it will consume less memory as fetched results are not
-     * remembered for future iterations.
-     *
-     * @param bool $enable Whether to enable buffering
-     * @return $this
-     */
-    public function enableBufferedResults(bool $enable = true)
```
Comment (markstory) [inline src/Database/Query.php:2184]: These methods look like good candidates for deprecation warnings in 4.x

**Comment by othercorey** [comment]
Looks good to me unless @ndm2 objects

File: src/Database/Query.php (hunk 2 above)
Comment (ADmad) [reply src/Database/Query.php:2184]: I have added a point to #16267 regarding adding deprecation to 4.x once we are doing with the changes in 5.0.

**Comment by ndm2** [comment]
No objections at this point, everything I've tested so far is still working fine.

**Review by ndm2 — APPROVED** [review]

**Pull request merged**